std/algorithm/searching.d

```diff
@@ -5116,17 +5154,45 @@ if (isInputRange!Range)
     void popFront()
     {
         assert(!empty, "Can not popFront of an empty Until");
-        if (!_openRight)
-        {
-            _done = predSatisfied();
-            _input.popFront();
-            _done = _done || _input.empty;
-        }
-        else
+        switch(_endType) with(EndType)
         {
-            _input.popFront();
-            _done = _input.empty || predSatisfied();
+            case sentinelIncluded:
+                static if(isInputRange!Sentinel)
+                {
+                    _input.popFront();
+                    if(_matchStarted)
+                    {
+                        _sentinel.popFront;
+                        _done = _input.empty || _sentinel.empty;
+                    }
+                    else
+                    {
+                        _matchStarted =  predSatisfied();
+                        _done = _sentinel.empty;
+                        if(_done)
+                        {
+                            _sentinel.popFront;
+                        }
+                    }
+                    break;
+                }
+                else
+                {
+                    assert(0,"Use firstPartOfSentinelIncluded with non-range sentinels");
+                }
```

Comment:
this doesn't appear to match the case the error message suggests it does. Also this should be 
`assert(isInputRange!Sentinel);` at the start of the case.

Reply:
I changed the code so I think this doesn't apply.